compile_executable.lua

```diff
@@ -0,0 +1,210 @@
+    return string.endswith(basename, LUA_EXT)
+end
+
+local function collect_lua_module_files(source_dir, modulename)
+    local module_files = {}
+
+    local module_dir = fio.pathjoin(source_dir, modulename)
+    local module_dir_files = get_directory_files(module_dir)
+
+    for _, file in ipairs(module_dir_files) do
+        if is_lua_file(file) then
+            table.insert(module_files, file)
+        end
+    end
+
+    local entrypoint_filename = string.format('%s.lua', modulename)
+    local entrypoint_filepath = fio.pathjoin(source_dir, entrypoint_filename)
+
+    if fio.path.exists(entrypoint_filepath) then
+        table.insert(module_files, entrypoint_filename)
+    end
+
+    return module_files
+end
+
+local function get_module_name(filepath)
+    -- aaa/bbb/ccc.lua -> aaa.bbb.bbb
+    -- aaa/bbb/init.lua -> aaa.bbb
+    local module_name
+    if fio.basename(filepath) == 'init.lua' then
+        local module_dir = fio.dirname(filepath)
+        module_name = table.concat(string.split(module_dir, '/'), '.')
+    else
+        local module_path = string.gsub(filepath, LUA_EXT, '')
+        module_name = table.concat(string.split(module_path, '/'), '.')
+    end
+
+    return module_name
+end
+
+local function generate_modules_preload(source_dir, files)
+    local res = {}
+
+    for _, filepath in ipairs(files) do
+        local module_name = get_module_name(filepath)
+
+        local full_filepath = fio.pathjoin(source_dir, filepath)
+        local module_content, err = read_file(full_filepath)
+        if module_content == nil then error(err) end
+
+        local module_preload = string.format(
+            MODULE_PRELOAD_TEMPLATE,
+            module_name,
+            module_name,
+            module_content
+        )
+        table.insert(res, module_preload)
+    end
+
+    return res
+end
+
+local function compile_executable(source_dir, dest_file, module_name)
+    source_dir = fio.abspath(source_dir)
+
+    local shebang = '#!/usr/bin/env tarantool\n'
+
+    local files = collect_lua_module_files(source_dir, module_name)
+    local modules_preload = generate_modules_preload(source_dir, files)
+
+    local module_run = string.format(MODULE_RUN_MAIN_TEMPLATE, module_name)
+
+    local executable_strings = merge_lists(
+        {shebang},
+        modules_preload,
+        {module_run}
+    )
+    local executable_content = table.concat(executable_strings, '\n')
+
+    local ok, err = write_file(dest_file, executable_content, tonumber('0755', 8))
+    if not ok then error(err) end
+end
+
+local args = argparse_internal.parse(arg, {
+    {'source', 'string'},
+    {'dest', 'string'},
+    {'module', 'string'},
+})
+
+assert(args.source ~= nil, "{lease, specify source directory (--source)")
+assert(args.dest ~= nil, "{lease, specify destination file name (--dest)")
+assert(args.module ~= nil, "{lease, specify module name (--module)")
+
+compile_executable(args.source, args.dest, args.module)
```

Comment:
One day we should carefully test this script

Reply:
Yes, but now I want to add a simple test that `tarantoolctl rocks make` creates working module.